### PR TITLE
fix: JavaScript translations

### PR DESCRIPTION
**Pull request by BryanttV**
### Description
This PR adds all configurations to use JavaScript translations.

### How To Test
1. Install this branch in your environment.
2. Change the platform language in `(Account → Site Preferences → Site Language)`. Set it to "Spanish (Latin America)"
3. Add a block in a unit from Studio.
4. You should see all the text of the block in the selected language.

### Using translation commands
1. Run `make extract_translations`
2. Add the translations in each `text.po` file
3. Run `make compile_translations`

### Screenshots
image
image
image

### CODE REVIEW

**Comment by mariajgrimaldi** [comment]
can you explain a bit more why it was failing?

File: mindmap/public/js/translations/es_ES/text.js
Comment (johnvente) [inline mindmap/public/js/translations/es_ES/text.js:1]: Could you format this file with prettier?

File: mindmap/public/js/translations/en/text.js
Comment (johnvente) [inline mindmap/public/js/translations/en/text.js:1]: Could you format this file with prettier?

File: mindmap/public/js/translations/es_419/text.js
Comment (johnvente) [inline mindmap/public/js/translations/es_419/text.js:1]: Same https://github.com/eduNEXT/xblock-mindmap/pull/20/files#r1327325285

File: mindmap/public/js/translations/es_ES/text.js
Comment (johnvente) [inline mindmap/public/js/translations/es_ES/text.js:1]: Same here https://github.com/eduNEXT/xblock-mindmap/pull/20/files#r1327325285

File: mindmap/public/js/translations/es_419/text.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+      return v;
+    }
+  };
+  
+
+  /* gettext library */
+
+  django.catalog = django.catalog || {};
+  
+  const newcatalog = {
```
Comment (johnvente) [inline mindmap/public/js/translations/es_419/text.js:27]: I recommend to use one file for this translations and each respective translation inside the same object:

something like this

```
 const newcatalog =  {
   "ES_419":{
      "Actions":"Acciones",
      "Back":"Volver"
   },
   "es_ES":{
      "Actions":"Acciones",
      "Back":"Volver"
   }
}
```

File: mindmap/public/js/translations/es_419/text.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    "\u2192 Click the circle to expand or collapse the child nodes.": "\u2192 Haga clic en el c\u00edrculo para expandir o contraer los nodos hijos.",
+    "\u2192 Ctrl + Enter: Create a new child node for the selected node.": "\u2192 Ctrl + Enter: Crear un nuevo nodo hijo para el nodo seleccionado.",
+    "\u2192 Delete/Supr: Delete the selected node.": "\u2192 Delete/Supr: Eliminar el nodo seleccionado.",
+    "\u2192 Double-click the node to edit it.": "\u2192 Doble clic en el nodo para editarlo.",
+    "\u2192 Drag the node to move it.": "\u2192 Arrastre el nodo para moverlo.",
+    "\u2192 Enter: Create a new brother node for the selected node.": "\u2192 Enter: Crear un nuevo nodo hermano para el nodo seleccionado.",
+    "\u2192 F2: Edit the selected node.": "\u2192 F2: Editar el nodo seleccionado.",
+    "\u2192 Space: Expand or collapse the selected node.": "\u2192 Espacio: Expandir o contraer el nodo seleccionado."
+  };
+  for (const key in newcatalog) {
```
Comment (johnvente) [inline mindmap/public/js/translations/es_419/text.js:78]: This for is not needed

django.catalog = {... newcatalog }

**Comment by BryanttV** [comment]
> Can you explain a bit more why it was failing?

It was failing because the `manage.py` file, it takes the settings defined in the `mindmap/settings/test.py` file, but this file did not have `statici18n` added to `INSTALLED_APPS`, and neither the necessary configuration:

```python
INSTALLED_APPS = (
    "statici18n" #  This allows to use the command
)

# statici18n
# https://django-statici18n.readthedocs.io/en/latest/settings.html

LANGUAGES = [
    ('en', 'English - Source Language'),
    ('es_419', 'Spanish (Latin America)'),
    ('es_ES', 'Spanish (Spain)'),
]

STATICI18N_DOMAIN = 'text'
STATICI18N_NAMESPACE = 'MindMapI18N'
STATICI18N_PACKAGES = (
    'mindmap',
)
STATICI18N_ROOT = 'mindmap/public/js'
STATICI18N_OUTPUT_DIR = 'translations'
```

Finally, the `compile_translations` command was updated to generate the JS files with the translations correctly:

```Makefile
compile_translations: 
    ...
    python manage.py compilejsi18n --namespace MindMapI18N --output $(JS_TARGET)
```

**Comment by BryanttV** [comment]
Hi @johnvente, the `text.js` files are automatically generated by the `compile_translations` command, so we would have to update those files every time the command is executed.

**Comment by johnvente** [comment]
> Hi @johnvente, the `text.js` files are automatically generated by the `compile_translations` command, so we would have to update those files every time the command is executed.

Hi @BryanttV in that case doesn't worth to update the files, it's ok for me

File: mindmap/public/js/src/mindmap.js [hunk 3]
```diff
@@ -7,6 +7,22 @@ function MindMapXBlock(runtime, element, context) {
   const removeGradeURL = runtime.handlerUrl(element, "remove_grade");
   const maxPointsAllowed = context.max_points;
 
+  var gettext;
```
[the remark this answers is not in the document]
Comment (BryanttV) [reply mindmap/public/js/src/mindmap.js]: Done

File: mindmap/public/js/src/mindmap.js [hunk 4]
```diff
@@ -7,6 +7,22 @@ function MindMapXBlock(runtime, element, context) {
   const removeGradeURL = runtime.handlerUrl(element, "remove_grade");
   const maxPointsAllowed = context.max_points;
 
+  let gettext;
+  if ("MindMapI18N" in window) {
```
Comment (johnvente) [inline mindmap/public/js/src/mindmap.js:11]:
```
if("MindMapI18N" in window ||  "gettext" in window) {
  gettext = window.MindMapI18N?.gettext || window.gettext;
}

if (typeof gettext == "undefined") {
    // No translations -- used by test environment
    gettext = (string) => string; 
  }
```

**Review by mariajgrimaldi — APPROVED** [review]
Thank you!

**Review by johnvente — APPROVED** [review]
Good job, LGTM!

**Pull request merged**